### PR TITLE
Fixes #3747 OAS3 validate param

### LINKED ISSUE #3747: OAS 3.0 | Swagger-UI | Required input parameters not getting validated
I downloaded the `Swagger-UI 3.X` code from https://github.com/swagger-api/swagger-ui as of 10/11/2017.

| Q                               | A
| ------------------------------- | -------
| Bug or feature request?         |  Bug
| Which Swagger/OpenAPI version?  | Open Api 3.0
| Which Swagger-UI version?       | swagger-ui, 3.x
| How did you install Swagger-UI? | Dowloaded zip, extracted the /dist folder and added my yaml file to it.
| Which browser & version?        | Chrome Version 61.0.3163.91
| Which operating system?         | macOS Sierrra

### Demonstration API definition

```yaml
openapi: "3.0.0"

info:
  version: 1.0.0
  title: Example API
  description: A sample API using OpenApi 3.0
  contact:
    name: ABC
    email: test@test.com

servers:
  - url: http://localhost:8080
    description: Local instance of API

paths:
  /examples/{id}:
    get:
      tags: [examples]
      description: Test path
      parameters:
      - name: id
        in: path
        required: true
        schema:
          type: string
      - name: test
        in: query
        description: test query
        required: false
        schema:
          type: string
          example: test
      responses:
        '200':
          description: test response
          content:
            application/json:
              schema:
                type: string
                example: success
```

Extracted the `dist` folder from swagger-UI and added the following`swaggerEx.yaml` file to it:

In the `index.html` folder I made the following change:
```
  // Build a system
  const ui = SwaggerUIBundle({
    url: "swaggerEx.yaml",
    dom_id: '#swagger-ui',
    deepLinking: true,
    presets: [
      SwaggerUIBundle.presets.apis,
      SwaggerUIStandalonePreset
    ],
    plugins: [
      SwaggerUIBundle.plugins.DownloadUrl
    ],
    layout: "StandaloneLayout"
  })

```

### Expected Behavior
As in swagger 2.0, I expected that the required parameters would be validated such as:

screen shot 2017-10-11 at 10 11 34 am

### Current Behavior
With the new UI I see that the `required` parameter validation does not occur and the call keeps `loading` but never gets executed. Here is a screen shot with that behavior:
screen shot 2017-10-11 at 10 14 15 am

### Context
We are trying to document our API using Swagger. Right now the documentation part works but if we want to provide user with the ability to try out the API we cannot do that because of this issue.

Thank you.

### PR DESCRIPTION
**Pull request by owenconti**
Fixes #3747 

Add flag for `isOAS3` on `validateParam` function. Access parameter details from OAS3's `schema` object.

Add a couple tests for new `isOAS3` flag.

### CODE REVIEW

File: test/core/utils.js [hunk 1]
```diff
@@ -274,464 +274,518 @@ describe("utils", function() {
...
-        value: ["test"],
-        items: {
-          type: "string"
-        }
-      })
-      result = validateParam( param, false )
-      expect( result ).toEqual( [] )
-    })
 
-    it("validates required booleans", function() {
```
Comment (owenconti) [inline test/core/utils.js]: Add new OAS3 spec tests here. Rest of the diff is because of indentation.

Comment (shockey) [reply test/core/utils.js]: https://github.com/swagger-api/swagger-ui/pull/3751/files?w=1 gives whitespace-free diffs 😉

Comment (owenconti) [reply test/core/utils.js]: should be the default!

**Comment by shockey** [comment]
Thanks @owenconti!

**Pull request merged**